src/background/assessment-data-converter.ts

```diff
@@ -183,11 +183,14 @@ export class AssessmentDataConverter {
         return null;
     }
 
-    public generateFailureInstance(description: string): UserCapturedInstance {
-        return {
+    public generateFailureInstance(description: string, path: string, snippet: string): UserCapturedInstance {
+        const instance = {
             id: this.generateUID(),
-            description: description,
-        };
+            ...(description !== '' && { description }),
+            ...(path !== '' && { selector: path }),
+            ...(snippet !== '' && { html: snippet }),
+        } as UserCapturedInstance;
```

Comment:
What's the purpose of doing the empty string checks here? Just curious as I'm not sure I'm following. 😄

Reply:
Currently, if a failure instance is added without a path or snippet, checking for the empty string ensure we don't add {path: ''} {snippet: ''} to our failure instance, and instead only add {comment: 'the comment'}.
This was to ensure that when users export reports, they dont see:
Comment            the comment
Path
Snippet
and instead just see the comment line. There are definitely other places in the code to put this check if you think it would be clearer elsewhere! Just wanted to make sure no work slipped out from beneath the feature flag before we were ready to release :)